base/compiler/abstractinterpretation.jl

```diff
@@ -1858,7 +1858,7 @@ function abstract_call_unionall(interp::AbstractInterpreter, argtypes::Vector{An
         a2 = argtypes[2]
         a3 = argtypes[3]
         ⊑ᵢ = ⊑(typeinf_lattice(interp))
-        nothrow = a2 ⊑ᵢ TypeVar && (a3 ⊑ᵢ Type || a3 ⊑ᵢ TypeVar)
+        nothrow = (!isvarargtype(a2) && a2 ⊑ᵢ TypeVar) && (!isvarargtype(a3) && (a3 ⊑ᵢ Type || a3 ⊑ᵢ TypeVar))
```

Comment:
`a2` can't be vararg, since it's not the last argtype. If it's vararg, we can never prove nothrow, but maybe still unwrap `a3` afterwards in case we can say something about the type?

Reply:
Ah right, `abstract_apply` truncates the first `Vararg` to ensure that `Vararg` only appears at the last argument: https://github.com/JuliaLang/julia/blob/7f37e709d36891b0ef263051c3df3b3529569f93/base/compiler/abstractinterpretation.jl#L1580-L1592